tests/test_sid.py

```diff
@@ -145,44 +160,52 @@ def _run_minibert(config, tmp_path, model_name, truncated, data_col_name: str =
+                                      truncated=truncated,
+                                      data_col_name=data_col_name,
+                                      num_threads=num_threads)
 
 
 @pytest.mark.slow
 @pytest.mark.use_cpp
 @pytest.mark.usefixtures("launch_mock_triton")
-def test_minibert_no_trunc(config, tmp_path):
+@pytest.mark.parametrize("num_threads", [1, 4])
+def test_minibert_no_trunc(config: Config, tmp_path: str, num_threads: int):
 
-    results = _run_minibert(config, tmp_path, "sid-minibert-onnx-no-trunc", False)
+    results = _run_minibert(config=config,
+                            tmp_path=tmp_path,
+                            model_name="sid-minibert-onnx-no-trunc",
+                            truncated=False,
+                            num_threads=num_threads)
 
-    # Not sure why these are different
-    if (CppConfig.get_should_use_cpp()):
-        assert results.diff_rows == 18
-    else:
-        assert results.diff_rows == 1333
+    # When threading is enabled, the results returned from the mocked Triton server won't match the expected results
+    if num_threads == 1:
+        # Not sure why these are different
+        if (CppConfig.get_should_use_cpp()):
+            assert results.diff_rows == 18
+        else:
+            assert results.diff_rows == 1333
 
 
 @pytest.mark.slow
 @pytest.mark.usefixtures("launch_mock_triton")
-def test_minibert_truncated(config, tmp_path):
-
-    results = _run_minibert(config, tmp_path, 'sid-minibert-onnx', True)
-
-    # Not sure why these are different
-    if (CppConfig.get_should_use_cpp()):
-        assert results.diff_rows == 1204
-    else:
-        assert results.diff_rows == 1333
-
-
-@pytest.mark.slow
-@pytest.mark.usefixtures("launch_mock_triton")
-def test_minibert_data_col_name(config, tmp_path):
-
-    results = _run_minibert(config, tmp_path, 'sid-minibert-onnx', True, "definitely_not_data")
-
-    # Not sure why these are different
-    if (CppConfig.get_should_use_cpp()):
-        assert results.diff_rows == 1204
-    else:
-        assert results.diff_rows == 1333
+@pytest.mark.parametrize("data_col_name", ["data", "definitely_not_data"])
+@pytest.mark.parametrize("num_threads", [1, 4])
+def test_minibert_truncated(config: Config, tmp_path: str, data_col_name: str, num_threads: int):
+
+    results = _run_minibert(config=config,
+                            tmp_path=tmp_path,
+                            model_name='sid-minibert-onnx',
+                            truncated=True,
+                            data_col_name=data_col_name,
+                            num_threads=num_threads)
+
+    # When threading is enabled, the results returned from the mocked Triton server won't match the expected results
+    if num_threads == 1:
+        # Not sure why these are different
+        if (CppConfig.get_should_use_cpp()):
+            assert results.diff_rows == 1204
+        else:
+            assert results.diff_rows == 1333
```

Comment:
We need to look into this more instead of codifying it into the tests. Why is it different?